README.md

```diff
@@ -1,28 +1,126 @@
 
-`train_neuralxc.py` is a Python script used to train an extreme classification model using **NeuralXC** which is an autoencoder-regression based model for this task.
+## Using the module
 
-## Setup
+### Setup
 
-Requirements:
+- Requirements for the project are listed in `requirements.txt`. In addition to these, it is mandatory to have PyTorch installed in your system with version 0.4.1 or higher.
+   - Requirements can be installed using `pip`:
+     ```bash
+     $ pip install -r requirements.txt 
+     ```
+     or using `conda`:
+     ```bash
+     $ conda install --file requirements.txt
+     ```
 
-- Python 2.7 or 3.5
-- PyTorch 0.4.1 or above
-- NumPy
-- SciPy
-- Matplotlib
-- Scikit-Learn
-- PyYAML
+### Installing `extreme_classification`
 
-Requirements can be installed using `pip`:
+- Clone the repository.
+
+- Run `python setup.py install`.
+
+- To test if your installation is successful, try running the command:
+```bash
+$ python -c "import extreme_classification"
+```
+
+## Using the training scripts
+
+- `train_neuralxc.py` is a Python script used to train an extreme classification model using **NeuralXC**. You can run the script by passing required options:
 ```bash
-$ pip install -r requirements.txt 
+python train_neuralxc.py [-h] --data_root DATA_ROOT --dataset_info DATASET_INFO
+                         --input_encoder_cfg INPUT_ENCODER_CFG --input_decoder_cfg INPUT_DECODER_CFG
+                         --output_encoder_cfg OUTPUT_ENCODER_CFG --output_decoder_cfg OUTPUT_DECODER_CFG
+                         --regressor_cfg REGRESSOR_CFG --optimizer_cfg OPTIMIZER_CFG
+                         [--init_scheme {xavier_uniform,kaiming_uniform,default}]
+                         [--device DEVICE] [--seed SEED] [--batch_size BATCH_SIZE] [--epochs EPOCHS]
+                         [--interval INTERVAL] [--input_ae_loss_weight INPUT_AE_LOSS_WEIGHT]
+                         [--output_ae_loss_weight OUTPUT_AE_LOSS_WEIGHT] [--k K]
+                         [--plot] [--save_model {all,inputAE,outputAE,regressor} [{all,inputAE,outputAE,regressor} ...]]
 ```
-or using `conda`:
+For more information about the options, please run `python train_neuralxc.py -h`.
+
+- `train_hierarchicalXC.py` is a Python script to train an extreme classification model using **HierarchicalXC**. You can run the script by passing required options:
 ```bash
-conda install --file requirements.txt
+python train_hierarchicalXC.py [-h] --data_root DATA_ROOT --dataset_info
+                               DATASET_INFO [--device DEVICE] [--seed SEED]
+                               [--batch_size BATCH_SIZE] [--epochs EPOCHS]
+                               [--interval INTERVAL] [--input_ae_dim INPUT_AE_DIM]
+                               [--njobs NJOBS] [--optimizer_cfg OPTIMIZER_CFG]
+                               [--save_model] [--k K]
 ```
+For more information about the options, please run `python train_hierarchicalxc.py -h`.
+
+
+### Configuration files

```

Comment:
Maybe this could be one level higher? (##)

Reply:
I suppose so. Thanks